src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/StreamExtensions.netstandard2.0.cs

```diff
@@ -50,5 +86,14 @@ internal static unsafe int Read(this Stream stream, byte* buffer, int size)
             int result = Interop.Kernel32.ReadFile(handle, buffer, size, out int bytesRead, IntPtr.Zero);
             return result == 0 ? 0 : bytesRead;
         }
+
+        internal static unsafe void ReadExactly(this Stream stream, byte* buffer, int size)
+        {
+            int bytesRead = stream is FileStream fs ? TryReadWin32File(fs, buffer, size) : 0;
+            if (bytesRead != size)
+            {
+                stream.CopyTo(buffer + bytesRead, size - bytesRead);
```

Comment:
When does this happen?

Reply:
We call `CopyTo` if we target a framework earlier than `$(NetCoreAppMinimum)` and either:

* `stream` is not a `FileStream`.
* We are not on Windows.
* We are on Windows but `ReadFile` read fewer bytes for some reason.

Reply:
> We call CopyTo if we target a framework earlier than $(NetCoreAppMinimum) and either:

So it's not really needed?

Reply:
It is needed, but only on downlevel frameworks. That's why I moved it to the `.netstandard2.0.cs` file.